### PR TITLE
Use CID to prevent incorrect usage in netmap

### LINKED ISSUE #216: Pivot for container placement
Why does `netmap` accept [any pivot](https://github.com/nspcc-dev/neofs-sdk-go/blob/d20999113a2ec57e6130aa264d818612af4283a2/netmap/netmap.go#L50)? 

In practice a node always [uses](https://github.com/nspcc-dev/neofs-node/blob/v0.28.0-rc.3/cmd/neofs-node/container.go#L358) `cid`. So maybe it would be better to change the interface to accept `cid` to prevent incorrect usage.

### PR DESCRIPTION
**Pull request by smallhive**
close #216

### CODE REVIEW

File: netmap/netmap.go [hunk 1]
```diff
@@ -166,9 +167,9 @@ func (m NetMap) PlacementVectors(vectors [][]NodeInfo, pivot []byte) ([][]NodeIn
 // the fixed NetMap and parameters.
 //
 // Result can be used in PlacementVectors.
-func (m NetMap) ContainerNodes(p PlacementPolicy, pivot []byte) ([][]NodeInfo, error) {
+func (m NetMap) ContainerNodes(p PlacementPolicy, containerID cid.ID) ([][]NodeInfo, error) {
 	c := newContext(m)
-	c.setPivot(pivot)
+	c.setPivot(containerID[:])
```
Comment (cthulhu-rider) [inline netmap/netmap.go:172]: right now correct, but lets use dedicated `Encode` method

**Comment by cthulhu-rider** [comment]
@smallhive btw how about https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go/netmap#NetMap.PlacementVectors? if u wanna refactor it in a separate PR then current one shouldn't close an issue.

**Review by cthulhu-rider — APPROVED** [review]

**Comment by smallhive** [comment]
Completed in this PR

File: netmap/netmap.go [hunk 2]
```diff
@@ -144,7 +146,10 @@ func flattenNodes(ns []nodes) nodes {
 // For example, in order to build node list to store the object, binary-encoded
 // object identifier can be used as pivot. Result is deterministic for
 // the fixed NetMap and parameters.
-func (m NetMap) PlacementVectors(vectors [][]NodeInfo, pivot []byte) ([][]NodeInfo, error) {
+func (m NetMap) PlacementVectors(vectors [][]NodeInfo, containerID cid.ID) ([][]NodeInfo, error) {
```
Comment (roman-khimov) [inline netmap/netmap.go:149]: It should be `oid` here. And comment needs to be updated.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
There is some problem with tests though.

**Comment by smallhive** [comment]
Yes, known issue. Me and @cthulhu-rider investigating it

**Comment by smallhive** [comment]
Tests problems were solved

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**